### PR TITLE
=str - Change implementation of Source.repeat to use unfold

### PR DESCRIPTION
**Pull request by viktorklang**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala [hunk 1]
```diff
@@ -237,14 +237,10 @@ object Source {
...
-  def repeat[T](element: T): Source[T, Unit] =
-    single(new immutable.Iterable[T] {
-      override val iterator: Iterator[T] = Iterator.continually(element)
-
-      override def toString: String = "repeat(" + element + ")"
-    })
-      .mapConcat(ConstantFun.scalaIdentityFunction)
-      .withAttributes(DefaultAttributes.repeat)
+  def repeat[T](element: T): Source[T, Unit] = {
+    val next = Some((element, element))
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: :microphone: *They see me cachin', they hatin'*

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: nice un.

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: nice :+1:

**Comment by ktoso** [comment]
LGTM, but let's start merging once I fix the build to include streams perhaps

**Comment by drewhk** [comment]
yes, validation does not yet build the streams stuff, so you cant trust it. It has to be rerun once the adjustments have been made.

**Comment by rkuhn** [comment]
LGTM

**Comment by johanandren** [comment]
LGTM

**Comment by rkuhn** [comment]
PLS BUILD

**Comment by rkuhn** [comment]
Thanks √!

**Pull request merged**